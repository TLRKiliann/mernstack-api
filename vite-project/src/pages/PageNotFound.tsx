import React from 'react'
import { Link } from 'react-router-dom'
import '../stylePages/PageNotFound.scss'

const PageNotFound: React.FC = () => {

	const colorErrorNotFound = {
		marginTop: '25%',
		textAlign: 'center',
		fontSize: '2rem',
		fontFamily: 'Junction, sans-serif',
		backgroundColor: '#242424',
		color: 'white'
	};

	return(
		<div style={colorErrorNotFound}>
			<h1>Error page not found !</h1>
			<Link to="/" >Redirect to home page.</Link>
		</div>
	)
}

export default PageNotFound