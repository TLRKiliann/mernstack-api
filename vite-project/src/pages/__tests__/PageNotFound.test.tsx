import React from "react"
import { test, expect, vi } from 'vitest'
import {fireEvent, render, screen} from '@testing-library/react'
import { Link } from 'react-router-dom'
import '@testing-library/jest-dom'
import { MemoryRouter } from 'react-router-dom'
import { create } from 'react-test-renderer'
import PageNotFound from "../PageNotFound.tsx"
//import userEvent from '@testing-library/user-event'
//import "@testing-library/jest-dom/extend-expect"
//import "@testing-library/jest-dom"
//import { act } from 'react-dom/test-utils'


test('PageNotFound testid', () => {
  render(    
    <MemoryRouter>
      <PageNotFound />
    </MemoryRouter>
  )
  const mytestid = screen.getByTestId("pagenotfoundtest")
  expect(mytestid).toBeInTheDocument()
})


test("PageNF Link test", () => {
  const pnfComp = create(
    <MemoryRouter>
      <PageNotFound />
    </MemoryRouter>
  )
  expect(pnfComp.toJSON()).toMatchSnapshot()
})
