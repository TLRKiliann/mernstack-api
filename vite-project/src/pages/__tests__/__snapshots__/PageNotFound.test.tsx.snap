// Vitest Snapshot v1

exports[`PageNF Link test 1`] = `
<div
  data-testid="pagenotfoundtest"
  style={
    {
      "backgroundColor": "#242424",
      "color": "white",
      "fontFamily": "Junction, sans-serif",
      "fontSize": "2rem",
      "marginTop": "25%",
      "textAlign": "center",
    }
  }
>
  <h1>
    Error page not found !
  </h1>
  <a
    href="/login"
    onClick={[Function]}
  >
    Redirect to home page.
  </a>
</div>
`;
